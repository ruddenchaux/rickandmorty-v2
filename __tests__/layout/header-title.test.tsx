import HeaderTitle from '@/app/layout/heder-title';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

jest.mock('next/navigation', () => ({
  usePathname: jest.fn().mockReturnValueOnce('/').mockReturnValueOnce('/path2'),
}));

jest.mock('../../app/layout/items-menu', () => ({
  itemsMenu: [
    { to: '/', label: 'Title 1' },
    { to: '/path2', label: 'Title 2' },
  ],
}));

afterEach(() => {
  jest.clearAllMocks();
});

describe('HeaderTitle', () => {
  it.each(['Title 1', 'Title 2'])(
    'should be rendered the title based on the pathname',
    (expectedTitle) => {
      const { container } = render(<HeaderTitle />);

      const title = screen.getByRole('heading', {
        name: `Rick and Morty - ${expectedTitle}`,
      });

      expect(title).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    }
  );
});
