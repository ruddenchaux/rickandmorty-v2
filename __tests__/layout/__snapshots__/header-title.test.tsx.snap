// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitle should be rendered the title based on the pathname 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-body1 css-1rpzg55-MuiTypography-root"
  >
    Rick and Morty - 
    Title 1
  </h1>
</div>
`;

exports[`HeaderTitle should be rendered the title based on the pathname 2`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-body1 css-1rpzg55-MuiTypography-root"
  >
    Rick and Morty - 
    Title 2
  </h1>
</div>
`;
